packages/workbox-background-sync/src/lib/request-manager.js

```
@@ -42,49 +42,60 @@ class RequestManager {
+        });
+        this._globalCallbacks.onResponse
+          && this._globalCallbacks.onResponse(hash, response);
+      }
+    } catch(err) {
+      this._globalCallbacks.onRetryFailure
+        && this._globalCallbacks.onRetryFailure(hash, err);
+      return Promise.reject(err);
+    }
+  }
+
+  /**
    * function to start playing requests
    * in sequence
-   * @return {void}
+   * @return {Promise}
    *
    * @memberOf RequestManager
    * @private
    */
-  replayRequests() {
-    return this._queue.queue.reduce((promise, hash) => {
-      return promise
-        .then(async (item) => {
-          const reqData = await this._queue.getRequestFromQueue({hash});
-          if(reqData.response) {
-            // check if request is not played already
-            return;
-          }
-
-          const request = await getFetchableRequest({
-            idbRequestObject: reqData.request,
-          });
-
-          return fetch(request)
-            .then((response)=>{
-              if(!response.ok) {
-                return Promise.resolve();
-              } else {
-                // not blocking on putResponse.
-                putResponse({
-                  hash,
-                  idbObject: reqData,
-                  response: response.clone(),
-                  idbQDb: this._queue.idbQDb,
-                });
-                this._globalCallbacks.onResponse
-                  && this._globalCallbacks.onResponse(hash, response);
-              }
-            })
-            .catch((err)=>{
-              this._globalCallbacks.onRetryFailure
-                && this._globalCallbacks.onRetryFailure(hash, err);
-            });
-        });
-    }, Promise.resolve());
+  async replayRequests() {
+    let failedItems = [];
+    for (let hash of this._queue.queue) {
+      try {
+        await this.replayRequest(hash);
+      } catch (err) {
+        failedItems.push(err);
+        failedItems++;
+      }
+    }
+    return failedItems.length > 0 ?
```

Since this is an async function, can you just throw an error?